pkg/platform/packet/packet.go

```diff
@@ -294,3 +330,117 @@ func (c *config) checkWorkerPoolNamesUnique() hcl.Diagnostics {
 
 	return diagnostics
 }
+
+func (c *config) checkReservationIDs() hcl.Diagnostics {
```

Comment:
It we want to be extra cautious, we should also check, that reservation IDs are not duplicated across the pools :see_no_evil:

Reply:
Terraform fails saying it is already in use... do you think it is really needed? I doubt, to be honest

Reply:
Well, IMO we should validate the configuration before even starting the configuration to follow "fail early" principle. But I don't think it's very important for now.

Reply:
I'm not sure I agree... this can open a can of worms and if we go down that route, start doing API calls to cloud providers to validate things (like if an DNS record already exists and will be possible to create it or fail because it exists).

If you want, I can create an issue to follow this later. But I'd like some guidance on what things you consider that should be catched, becase of course doing it for "all possible issues" can create the problem just mentioned.